specification/storagetasks/resource-manager/Microsoft.StorageTasks/stable/2023-01-01/storagetasks.json

```diff
@@ -0,0 +1,903 @@
+      "properties": {
+        "operations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageTaskOperation"
+          },
+          "x-ms-identifiers": [],
+          "description": "List of operations to execute in the else block"
+        }
+      },
+      "required": [
+        "operations"
+      ],
+      "description": "The else block of storage task operation"
+    },
+    "StorageTaskOperation": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The operation to be performed on the object.",
+          "enum": [
+            "SetBlobTier",
+            "SetBlobTags",
+            "SetBlobImmutabilityPolicy",
+            "SetBlobLegalHold",
+            "SetBlobExpiry",
+            "DeleteBlob",
+            "UndeleteBlob"
+          ],
+          "x-ms-enum": {
+            "name": "StorageTaskOperation",
+            "modelAsString": true
+          }
+        },
+        "parameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Key-value parameters for the operation."
+        },
+        "onSuccess": {
+          "type": "string",
+          "description": "Action to be taken when the operation is successful for a object.",
+          "enum": [
+            "continue"
+          ],
+          "x-ms-enum": {
+            "name": "onSuccess",
+            "modelAsString": false
+          }
+        },
+        "onFailure": {
+          "type": "string",
+          "description": "Action to be taken when the operation fails for a object.",
+          "enum": [
+            "break"
+          ],
+          "x-ms-enum": {
+            "name": "onFailure",
+            "modelAsString": false
+          }
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "Represents an operation to be performed on the object"
+    },
+    "StorageTasksListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageTask"
+          },
+          "x-ms-identifiers": [],
+          "description": "Gets the list of storage tasks and their properties."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Request URL that can be used to query next page of storage tasks. Returned when total number of requested storage tasks exceed maximum page size.",
+          "readOnly": true
+        }
+      },
+      "description": "The response from the List Storage Tasks operation."
+    },
+    "StorageTaskAssignmentsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageTaskAssignmentId"
```

Comment:
>StorageTaskAssignmentId [](http://example.com/codeflow?start=35&length=23)

This should just be called "StorageTaskAssignment" instead of "StorageTaskAssignmentId", as later it can have more properties instead of just "id".

Reply:
yes makes sense, made the change